### PR TITLE
Prevent & log NPE in KubernetesRuntime;

### PR DESCRIPTION
**Pull request by mshaposhnik**
### What does this PR do?
Add more debugging and prevent NPE in case machine can not be found by it's name in configs map.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/15974

#### Docs PR
N/A

### CODE REVIEW

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java [hunk 1]
```diff
@@ -814,6 +814,17 @@ private void storeStartingMachine(
       String machineName = Names.machineName(toCreateMeta, container);
 
       LOG.debug("Creating machine '{}' in workspace '{}'", machineName, workspaceId);
+      // Sometimes we facing NPE trying to retrieve machine config. Possible names mismatch. Need to get more info on that cases.
+      Optional<InternalMachineConfig> machineConfigOpt = Optional
```
Comment (metlos) [inline infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java]: Why not a simple null-check?

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java [hunk 2]
```diff
@@ -814,6 +814,17 @@ private void storeStartingMachine(
...
+          .ofNullable(machineConfigs.get(machineName));
+      if (!machineConfigOpt.isPresent()) {
+        LOG.error(
+            "Workspace '{}' start failed. Machine with name '{}' requested but not found in configs map. Present machines are: {}.",
+            workspaceId, machineName, String.join(",", machineConfigs.keySet()));
+        throw new InfrastructureException(format(
+            "Unable to start workspace '%s' due to internal error. Case will be investigated.",
```
Comment (metlos) [inline infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java]: IMHO this is not very informative for the users. This error will end up in the startup log in the UI, so it should contain at least a little bit of info on how to proceed from there.
I would personally formulate the message along the lines of: `Unable to start the workspace '%s' due to an internal inconsistency while composing the workspace runtime. Please report a bug with the details contained in the log of the Che server.`

Comment (mshaposhnik) [reply infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java]: most users cannot see log of Che server :)

Comment (sparkoo) [reply infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java]: that is probably true. What about printing the `workspaceId`, `machineName` and `machineConfigs` and write message to report the bug with this info? Is it safe to print this internal stuff to end-user?

Btw original sentence "Case will be investigated." sounds like some data were sent somewhere and we're looking at it. I'm more inclined to Lukas's version, but maybe end it with something like `Please report a bug. If possible, include the details from Che server log in bug report (your admin can help with that).`

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java (hunk 1 above)
Comment (sparkoo) [reply infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java]: or 
```
InternalMachineConfig machineConfig = Optional.ofNullable(...).orElseThrow(() -> {
  LOG.error(...)
  return new InfrastructureException(...)
})
```

Comment (mshaposhnik) [reply infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java]: yup that;s better

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java (hunk 2 above)
Comment (mshaposhnik) [reply infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java]: Ok recomposed

**Comment by mshaposhnik** [comment]
crw-ci-test

**Review by sparkoo — APPROVED** [review]

**Comment by dmytro-ndp** [comment]
@mshaposhnik: [Happy path tests are failing](https://github.com/eclipse/che/pull/16580#issuecomment-611959185) because they are outdated in this PR. Please, take latest changes from **master** branch.

**Comment by mshaposhnik** [comment]
@dmytro-ndp thanks

**Review by skabashnyuk — APPROVED** [review]

**Pull request merged**